pandas/tests/indexes/datetimes/test_ops.py

```diff
@@ -460,6 +460,20 @@ def test_add_dti_dti(self):
         with pytest.raises(TypeError):
             dti + dti_tz
 
+    def test_add_dti_ndarray(self):
+        # GH 17558
+        # Check that tz-aware DatetimeIndex + np.array(dtype="timedelta64")
+        # and DatetimeIndex + TimedeltaIndex work as expected
+        dti = pd.DatetimeIndex([pd.Timestamp("2017/01/01")])
+        dti = dti.tz_localize('US/Eastern')
+        expected = pd.DatetimeIndex([pd.Timestamp("2017/01/01 01:00")])
+        expected = expected.tz_localize('US/Eastern')
+
+        td_np = np.array([np.timedelta64(1, 'h')], dtype="timedelta64[ns]")
+        tm.assert_index_equal(dti + td_np, expected)
+        tm.assert_index_equal(dti + td_np[0], expected)
+        tm.assert_index_equal(dti + TimedeltaIndex(td_np), expected)
```

Comment:
Can we just for-loop these checks instead?  Also, let's be a little more explicit and write it this way:
~~~python
actual = dti + ...
tm.assert_index_equal(actual, expected)
~~~